django_project/clip-and-ship/views.py

```diff
@@ -79,24 +130,51 @@ def clip_layer(request, layername):
         y = float(bbox_array[3]) - float(bbox_array[1])
         height = int(y * 43260) + offset
 
-        wcs_formatted_url = wcs_url.format(
-            layer_name=layername,
-            bbox=bbox_string,
-            width=width,
-            height=height
-        )
+        width = int(width * settings.WCS_DOWNLOADED_RATIO_SIZE)
+        height = int(height * settings.WCS_DOWNLOADED_RATIO_SIZE)
 
         extention = 'tif'
 
-        raster_filepath = os.path.join(
-                temporary_folder,
-                layer.title + '.' + extention)
+        # checking file size of wcs download
+        # it is done by assumption by getting sample
+        # 1. getting part 10x10 through wcs
+        # 2. get the filesize
+        # 3. times filesize by width/10 * height/10, as assumption for actual filesize
+        # -------------------------------------------------
+        width_sample = int(width / SIZE_SAMPLE)
+        height_sample = int(height / SIZE_SAMPLE)
 
-        response = urllib2.urlopen(wcs_formatted_url)
+        sample_filepath = os.path.join(
+            temporary_folder,
+            layer.title + '.sample.' + extention)
+        download_wcs(
+            layername, bbox_string,
+            SIZE_SAMPLE,
```

Comment:
shouldn't you be using width_sample here?

Reply:
hi @cchristelis 
the idea is size simple is the sample of size, which are 100*100 in the setting
width_sample is the how many width for 100*100 for giving actual width
i think it is naming, i will update it